### PR TITLE
Gherkin: Consolidating core dashboard with dev perspective- Monitoring

### PR DESCRIPTION
**Pull request by sanketpathak**
**Epic**: https://issues.redhat.com/browse/ODC-3698
**Story**: https://issues.redhat.com/browse/ODC-5966

**Acceptance criteria:**

- Update scripts based on the following changes:
    - Monitoring area should be renamed to Observe
    - Namespace selector shouldn't be present
    - Only 4 items should be in the Dashboard selector
        - Kubernetes / Compute Resources / Namespace (Workloads)
        - Kubernetes / Compute Resources / Namespace (Pods)
        - Kubernetes / Compute Resources / Workloads
        - Kubernetes / Compute Resources / Pods 

**Checks required for approving Epic gherkin scripts PR:**

- [x] Add @epic-number to the scenarios or feature file [e.g: @odc-xxx]
- [x] Add @to-do for automation possible scenarios
- [x] Add @regression or @smoke based on the importance of the scenario
- [x] Update the test scenarios count in [Automation status confluence Report](https://docs.jboss.org/display/ODC/Automation+Status+Report)
- [ ] Check for the linter issues by executing `yarn run gherkin-lint` on frontend folder [Skip epic number tags related linter issues]

### CODE REVIEW

**Review by gajanan-more — APPROVED** [review]
/lgtm

**Review by jerolimov — CHANGES_REQUESTED** [review]
Why is gherkin lint doesn't blame these new `@observe` annotations when is not defined in `.gerkin-lintrc`

See https://github.com/openshift/console/blob/master/frontend/packages/dev-console/integration-tests/.gherkin-lintrc#L15

And after that: Shouldn't you replace the annotation there as well?

I'm unsure if we really want rename the annotations since the frontend module and also the feature file still contains monitoring in their names.

File: frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature [hunk 1]
```diff
@@ -1,67 +1,59 @@
...
-              And user will see Receive Bandwidth on Dashboard tab
-              And user will see Transmit Bandwidth on Dashboard tab
-              And user will see the Rate of Received Packets on Dashboard tab
-              And user will see the Rate of Transmitted Packets on Dashboard tab
-              And user will see the Rate of Received Packets Dropped on Dashboard tab
-              And user will see the Rate of Transmitted Packets Dropped on Dashboard tab
+             Then user will see the "Namespace (Workloads)" on Dashboard tab
+              And user will see the "Namespace (Pods)" on Dashboard tab
+              And user will see "Workloads" on Dashboard tab
+              And user will see "Pods" on Dashboard tab
```
Comment (vikram-raj) [inline frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature:29]: This scenario is changed. Now user can select the workload after selecting the `Kubernetes / Compute Resources / workload` dashboard.

File: frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature [hunk 2]
```diff
@@ -1,67 +1,59 @@
...
-              And user will see Receive Bandwidth on Dashboard tab
-              And user will see Transmit Bandwidth on Dashboard tab
-              And user will see the Rate of Received Packets on Dashboard tab
-              And user will see the Rate of Transmitted Packets on Dashboard tab
-              And user will see the Rate of Received Packets Dropped on Dashboard tab
-              And user will see the Rate of Transmitted Packets Dropped on Dashboard tab
+              And user will see the "Namespace (Workloads)" on Dashboard tab
+              And user will see the "Namespace (Pods)" on Dashboard tab
+              And user will see "Workloads" on Dashboard tab
+              And user will see "Pods" on Dashboard tab
```
Comment (vikram-raj) [inline frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature:43]: Now all workloads dropdown will be not visible upfront. And user will see these options in the dashboard dropdown.

File: frontend/packages/dev-console/integration-tests/features/monitoring/topology-sidebar-actions.feature [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
-              And user will see Memory Usage Metrics
-              And user will see Receive Bandwidth Metrics
-              And user will see All Events dropdown
+              And user clicks on Observe tab
+              And user clicks on View monitoring dashboard link
+             Then page redirected to the Observe page
+              And user will see the "Namespace (Workloads)" on Dashboard tab
+              And user will see the "Namespace (Pods)" on Dashboard tab
+              And user will see "Workloads" on Dashboard tab
+              And user will see "Pods" on Dashboard tab
```
Comment (vikram-raj) [inline frontend/packages/dev-console/integration-tests/features/monitoring/topology-sidebar-actions.feature:22]: These are the options of the `Dashboards` dropdown.

**Review by vikram-raj — COMMENTED** [review]
The scenario of selecting the dashboard is missing.

File: frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature (hunk 2 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature (hunk 1 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature]: Updated

**Review by vikram-raj — COMMENTED** [review]
/lgtm

File: frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature [hunk 4]
```diff
@@ -1,67 +1,115 @@
...
+              And user clicks on "Kubernetes / Compute Resources / Namespace (Workloads)" option
+              And user clicks on Type dropdown
+              And user selects "deployment" option from the dropdown
+             Then user will see "CPU Usage" chart
+              And user will see "CPU Quota" chart
+              And user will see "Memory Usage" chart
+              And user will see "Memory Quota" chart
+              And user will see "Current Network Usage" chart
+              And user will see "Bandwidth" charts
+              And user will see "Rate of Packets" charts
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature:33]:
```suggestion
              And user will see "Rate of Packets" chart
```

File: frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature [hunk 5]
```diff
@@ -1,67 +1,115 @@
...
+              And user will see "Rate of Packets Dropped" charts
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature:34]:
```suggestion
              And user will see "Rate of Packets Dropped" chart
```

File: frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature [hunk 6]
```diff
@@ -1,67 +1,115 @@
...
+              And user will see "Memory Utilisation (from requests)" chart
+              And user will see "Memory Utilisation (from limits)" chart
+              And user will see "CPU Usage" chart
+              And user will see "CPU Quota" chart
+              And user will see "Memory Usage" chart
+              And user will see "Memory Quota" chart
+              And user will see "Current Network Usage" chart
+              And user will see "Bandwidth" charts
+              And user will see "Rate of Packets" charts
+              And user will see "Rate of Packets Dropped" charts
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature:54]: Modify the charts to chart

File: frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature [hunk 7]
```diff
@@ -1,67 +1,115 @@
...
+              And user selects "deployment" option from the dropdown
+             Then user will see "CPU Usage" chart
+              And user will see "CPU Quota" chart
+              And user will see "Memory Usage" chart
+              And user will see "Memory Quota" chart
+              And user will see "Current Network Usage" chart
+              And user will see "Bandwidth" charts
+              And user will see "Average Container Bandwidth by Pod" charts
+              And user will see "Rate of Packets" charts
+              And user will see "Rate of Packets Dropped" charts
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature:77]: same as above

File: frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature [hunk 8]
```diff
@@ -1,67 +1,115 @@
...
+             Then user will see "CPU Usage" chart
+              And user will see "CPU Throttling" chart
+              And user will see "CPU Quota" chart
+              And user will see "Memory Usage" chart
+              And user will see "Memory Quota" chart
+              And user will see "Bandwidth" charts
+              And user will see "Rate of Packets" charts
+              And user will see "Rate of Packets Dropped" charts
+              And user will see "Storage IO- Distribution(Pod - Read & Writes)" charts
+              And user will see "Storage IO - Distribution(Containers)" charts
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature:98]: same as above

File: frontend/packages/dev-console/integration-tests/features/monitoring/topology-sidebar-actions.feature [hunk 9]
```diff
@@ -1,25 +1,24 @@
...
             Given user is at developer perspective
               And user is at Add page
-              And user has created or selected namespace "aut-monitoring-sidebar"
+              And user has created or selected namespace "aut-observe-sidebar"
 
 
         @smoke
-        Scenario Outline: Navigating to Monitoring page from topology page Sidebar for "<resourceType>" workload : M-04-TC01
+        Scenario Outline: Navigating to Observe page from topology page Sidebar for "<resourceType>" workload : M-04-TC01
             Given workload "<workload>" with resource type "<resourceType>" is present in topology page
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/monitoring/topology-sidebar-actions.feature:14]:
```suggestion
            Given user has created workload "<workload>" with resource type "<resourceType>"
            And user is at Topology page
```

**Review by makambalaji — CHANGES_REQUESTED** [review]
Small modifications are needed and lets wait for the [comment](https://issues.redhat.com/browse/ODC-3698?focusedCommentId=16494404&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16494404) to resolve

File: frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature (hunk 4 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature:33]: There are more than one chart

File: frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature (hunk 6 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature:54]: More than one chart appearing for Rate of Packets Dropped

File: frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature (hunk 7 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature:77]: Same as above

File: frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature (hunk 8 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature:98]: Same as above

File: frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature (hunk 5 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature:34]: Same as above

File: frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature [hunk 10]
```diff
@@ -1,67 +1,125 @@
...
+              And user will see "Storage IO - Distribution" chart
+
+
+        @regression, @odc-3698
+        Scenario: Inspect chart directs to Metrics tab: M-02-TC06
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature:103]: @makambalaji Please review this scenario as well. cc: @vikram-raj

File: frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature [hunk 11]
```diff
@@ -1,67 +1,125 @@
...
-              And user will see the Memory Usage on Dashboard tab
-              And user will see Receive Bandwidth on Dashboard tab
-              And user will see Transmit Bandwidth on Dashboard tab
-              And user will see the Rate of Received Packets on Dashboard tab
-              And user will see the Rate of Transmitted Packets on Dashboard tab
-              And user will see the Rate of Received Packets Dropped on Dashboard tab
-              And user will see the Rate of Transmitted Packets Dropped on Dashboard tab
+              And user clicks on Dashboard dropdown
+              And user clicks on "Kubernetes / Compute Resources / Namespace (Pods)" option
+              And user clicks on Inspect on "CPU Usage" chart
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature:108]:
```suggestion
              And user clicks Inspect on "CPU Usage" chart
```

File: frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature [hunk 12]
```diff
@@ -1,67 +1,125 @@
...
+             Then user will see Metrics tab in Observe is open with "CPU Usage" option selected
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature:109]:
```suggestion
             Then user will see Metrics tab in Observe page
             And "CPU Usage" option selected by default
```

Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/monitoring/dashboard-display.feature]: Updated

**Comment by makambalaji** [comment]
/lgtm
Added a comment in epic, If there is any change due to that comment. Will raise a new pr

**Comment by jerolimov** [comment]
PR addresses integration tests only.

/label px-approved
/label docs-approved
/label qe-approved

**Pull request merged**